backend/src/main/java/com/bakdata/conquery/sql/conversion/cqelement/CQConceptConverter.java

```diff
@@ -34,50 +33,172 @@ public Class<CQConcept> getConversionClass() {
+		ConceptSelects.ConceptSelectsBuilder selectsBuilder = ((ConceptSelects) previous.getQualifiedSelects()).toBuilder();
+		selectsBuilder.dateRestriction(Optional.empty());
+		if (node.isExcludeFromTimeAggregation()) {
+			selectsBuilder.validityDate(Optional.empty());
+		}
+		return selectsBuilder.build();
+	}
+
+	private List<Condition> buildDateRestriction(ConversionContext context, QueryStep previous) {
+		return ((ConceptSelects) previous.getSelects()).getDateRestriction()
+													   .map(dateRestrictionColumn -> getDateRestrictionAsCondition(context, previous, dateRestrictionColumn))
+													   .orElseGet(Collections::emptyList);
+	}
+
+	private static List<Condition> getDateRestrictionAsCondition(ConversionContext context, QueryStep previous, Field<Object> dateRestrictionColumn) {
+		return previous.getSelects().getValidityDate().stream()
+					   .map(validityDateColumn -> context.getSqlDialect().getFunction().dateRestriction(dateRestrictionColumn, validityDateColumn))
+					   .toList();
+	}
+
+	private ConceptSelects prepareEventSelectSelects(
+			ConversionContext context,
+			CQTable table,
+			QueryStep previous
+	) {
+		return ((ConceptSelects) previous.getQualifiedSelects()).withEventSelect(this.getEventSelects(context, table));
+	}
+
+	private ConceptSelects prepareEventFilterSelects(QueryStep previous) {
+		return ((ConceptSelects) previous.getQualifiedSelects()).withEventFilter(List.of());
+	}
+
+	private List<Condition> buildEventFilterConditions(ConversionContext context, CQTable table) {
 		return table.getFilters().stream()
-					.map(filter -> filterConverterService.convert(filter, context))
+					.map(filterValue -> this.filterConverterService.convert(filterValue, context))
 					.toList();
 	}
 
-	private String getPrimaryColumnName(CQTable table) {
-		// TODO add primary column to table
-		return "pid";
+	private List<Field<Object>> getEventSelects(ConversionContext context, CQTable table) {
+		return table.getSelects().stream()
+					.map(select -> (Field<Object>) this.selectConverterService.convert(select, context))
+					.toList();
 	}
 
-	private List<Field<?>> getConceptSelects(CQTable table, ConversionContext context) {
-		return table.getSelects().stream()
-					.map(select -> selectConverterService.convert(select, context))
-					.collect(Collectors.toList());
+	/**
+	 * selects:
+	 * - all of previous step
+	 */
+	private QueryStep buildFinalQueryStep(String conceptLabel, QueryStep previous) {
+		ConceptSelects finalSelects = ((ConceptSelects) previous.getQualifiedSelects());
+		return QueryStep.builder()
+						.cteName(createCteName(conceptLabel, ""))
+						.fromTable(QueryStep.toTableLike(previous.getCteName()))
+						.selects(finalSelects)
+						.conditions(List.of())
+						.predecessors(List.of(previous))
+						.build();
+	}
+
+	private static String createCteName(String conceptLabel, String suffix) {
```

Comment:
Ich glaube hier müsst ihr noch Query-globale disambiguation machen, dass dann AND auf dem gleichen Concept möglich ist

Reply:
Das haben wir erstmal mit einem inkrementierenden Counter implementiert, siehe a178d3e44ab36be2278c5f6091894270a2c381ed

Reply:
Habs schon gesehen, danke auch für den Test